test/compare_outputs.py

```diff
@@ -0,0 +1,417 @@
+        """
+        image = FITSImage(fname)
+        image.calc_noise()
+        return image.noise
+
+    logger.debug("Comparing noise in images '%s' and '%s'", file1, file2)
+    noise1 = image_noise(file1)
+    noise2 = image_noise(file2)
+    agree = np.isclose(noise1, noise2, rtol=rtol)
+    if agree:
+        logger.info("Image files '%s' and '%s' are similar", file1, file2)
+    if not agree:
+        if verbosity > 0:
+            logger.error(
+                "Image files '%s' and '%s' differ (rtol=%s)\n"
+                "Noise: %.3f mJy/b vs %.3f mJy/b",
+                file1,
+                file2,
+                rtol,
+                1000 * noise1,
+                1000 * noise2,
+            )
+        else:
+            logger.error(
+                "Image files '%s' and '%s' differ (rtol=%s)", file1, file2, rtol
+            )
+    return agree
+
+
+def compare_results(dcmp, rtol, verbosity, check_images=True, check_catalogs=False, check_other=False):
+    """
+    Compare the results between the files produced by each PyBDSF run.
+    :param dircmp dcmp: the directory contents of the two workflows
+    :param float rtol: relative tolerance threshold
+    :param int verbosity: verbosity level, higher means more verbose
+    """
+    # Initialize logger and set log level depending on verbosity level
+    init_logger(verbosity)
+
+    logger.info("*** Comparing results in '%s' and '%s' ***", dcmp.left, dcmp.right)
+    logger.info(f"Checking images: {check_images}")
+    logger.info(f"Checking catalogs: {check_catalogs}")
+    logger.info(f"Checking other: {check_other}")
+    agree = True
+    for dname, sub_dcmp in dcmp.subdirs.items():
+        logger.debug(
+            "dname: %s, sub_dcmp.left: %s, sub_dcmp.right: %s",
+            dname,
+            sub_dcmp.left,
+            sub_dcmp.right,
+        )
+        check_all_files_present(sub_dcmp)
+        for fname in sub_dcmp.common_files:
+            left = os.path.join(sub_dcmp.left, fname)
+            right = os.path.join(sub_dcmp.right, fname)
+            root, ext = os.path.splitext(fname)
+            logger.debug("fname: %s, root: %s, ext: %s", fname, root, ext)
+            if check_images:
+                if "fits" in ext.lower() and "catalogues" not in dname:
+                    if not image_files_are_similar(
+                        left, right, rtol=rtol, verbosity=verbosity
+                    ):
+                        agree = False
+            if check_catalogs:
+                if "fits" in ext.lower() and "catalogues" in dname:
+                    if not fits_files_are_similar(
+                        left, right, rtol=rtol, verbosity=verbosity
+                    ):
+                        agree = False
+            if check_other:
+                if not text_files_are_similar(
+                    left, right, verbosity=verbosity
+                ):
+                    agree = False
+    return agree
+
+
+class FITSImage(object):
```

Comment:
Please add a note that this class was copied (almost) verbatim from the Rapthor repository: `rapthor/lib/fitsimage.py`. Maybe add a hyperlink to this file on GitLab as well?

Reply:
Done.